pkg/domain/entities/types/images.go

```diff
@@ -147,6 +147,57 @@ type ImagePullReport struct {
 	ID string `json:"id,omitempty"`
 }
 
+// ImagePullReportV2 provides a detailed status of a container image pull operation.
+type ImagePullReportV2 struct {
+	// Status indicates the current state of the image pull operation.
```

Comment:
“indicates the meaning of the current progress entry”.

Important: It is explicitly allowed for future versions to add new kinds of entries; consumers should ignore entries with unknown "status".